### PR TITLE
Stop reporting/processing the number of mixing participants in DSSTATUSUPDATE

### PR DESCRIPTION
**Pull request by UdjinM6**

### CODE REVIEW

**Comment by codablock** [comment]
I'd prefer to move the deserialization of DSSTATUSUPDATE into its own message (e.g. `CPrivateSendStatusUpdate` and then also add protocol version handling code to completely omit `nMsgEntriesCount` when proto version is recent. This would later allow to completely remove that member.

**Comment by UdjinM6** [comment]
Good idea! See f37f545.

File: src/privatesend/privatesend.h [hunk 1]
```diff
@@ -53,27 +53,64 @@ enum PoolMessage {
...
+        nMessageID(nMessageID) {};
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action)
+    {
+        READWRITE(nSessionID);
+        READWRITE(nState);
+        if (s.GetVersion() <= 702015) {
```
Comment (codablock) [inline src/privatesend/privatesend.h]: Do you want to do the proto bump as part of this PR or another one?

File: src/privatesend/privatesend.h [hunk 2]
```diff
@@ -53,27 +53,64 @@ enum PoolMessage {
...
+public:
+    int nSessionID;
+    PoolState nState;
+    int nEntriesCount; // deprecated, kept for backwards compatibility
+    PoolStatusUpdate nStatusUpdate;
+    PoolMessage nMessageID;
+
+    CPrivateSendStatusUpdate() :
+        nSessionID(0),
+        nState(POOL_STATE_IDLE),
```
Comment (codablock) [inline src/privatesend/privatesend.h]: BTW, I'm a big fan of [default initializers for class members](https://arne-mertz.de/2015/08/new-c-features-default-initializers-for-member-variables/), which IMHO gives much easier to read code and provides much more secure code (introducing a new constructor does not by accident leave out member initialization).

We don't have this as part of our code style guidelines, but maybe we should introduce it (assuming you also like the new style of initialization). Obviously, we won't easily adopt that style to old Bitcoin code...but doing it in our own code is at least something...

File: src/privatesend/privatesend.h (hunk 1 above)
Comment (UdjinM6) [reply src/privatesend/privatesend.h]: Changes here are backwards compatible, so there is no need for a protobump for this specific change imo. We can do it later when smth new/incompatible is introduced.

Comment (UdjinM6) [reply src/privatesend/privatesend.h]: Changes here are backwards compatible, so there is no need for a protobump for this specific case imo. We can do it later when smth new/incompatible is introduced.

File: src/privatesend/privatesend.h [hunk 3]
```diff
@@ -53,27 +53,64 @@ enum PoolMessage {
...
 // pool states
-enum PoolState {
+enum PoolState : int32_t {
     POOL_STATE_IDLE,
     POOL_STATE_QUEUE,
     POOL_STATE_ACCEPTING_ENTRIES,
     POOL_STATE_SIGNING,
     POOL_STATE_ERROR,
-    POOL_STATE_SUCCESS,
-    POOL_STATE_MIN = POOL_STATE_IDLE,
```
Comment (codablock) [inline src/privatesend/privatesend.h]: Please note that the serialization code does not check for valid enum values...it does a plain and stupid cast to the enum type. So, validity/range checks must still be done (e.g. as part of SerializationOp)

**Comment by PastaPastaPasta** [comment]
Not sure how this fixes the issue addressed in 150d40deb4b786ac727fba158b09e7d2d5913c51

**Comment by UdjinM6** [comment]
@PastaPastaPasta it doesn't address the one for the logic you implemented, it addresses a similar legacy one.

File: src/privatesend/privatesend-server.cpp [hunk 4]
```diff
@@ -845,7 +845,7 @@ void CPrivateSendServer::PushStatus(CNode* pnode, PoolStatusUpdate nStatusUpdate
 {
     if (!pnode) return;
     CNetMsgMaker msgMaker(pnode->GetSendVersion());
-    connman.PushMessage(pnode, msgMaker.Make(NetMsgType::DSSTATUSUPDATE, nSessionID, (int)nState, (int)vecEntries.size(), (int)nStatusUpdate, (int)nMessageID));
+    connman.PushMessage(pnode, msgMaker.Make(NetMsgType::DSSTATUSUPDATE, nSessionID, (int)nState, 0, (int)nStatusUpdate, (int)nMessageID));
```
Comment (PastaPastaPasta) [inline src/privatesend/privatesend-server.cpp]: maybe put a TODO or something somewhere to remind ourselves we should remove this fully at some point and do a proto bump?

Comment (codablock) [reply src/privatesend/privatesend-server.cpp]: This is an outdated change and was later replaced with CPrivateSendStatusUpdate

Comment (PastaPastaPasta) [reply src/privatesend/privatesend-server.cpp]: :+1:

**Pull request merged**